### PR TITLE
Fix/10886/laggy search

### PR DESCRIPTION
**Pull request by osmaczko**
### What does the PR do

closes: https://github.com/status-im/status-desktop/issues/10886

status-go PR: https://github.com/status-im/status-go/pull/3586

check by commits, in short:
- use mediaserver for contact images
- optimise the usage of `getContactsByGroup`
- decrease MESSAGES_PER_PAGE_MAX to avoid processing too many messages at once

### Affected areas

- displaying images, especially contact images
- mentions in messages
- contact groups

### Before

https://github.com/status-im/status-desktop/assets/33099791/dbb34e2f-dfc8-446a-8398-b241e1da1139

### After

https://github.com/status-im/status-desktop/assets/33099791/99ccc87a-33f8-4234-a66e-daf7dee930da

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml [hunk 1]
```diff
@@ -33,10 +33,14 @@ QtObject {
...
-    property bool isImage: false
+    property bool isImage: isImgSrc(root.name)
     property int imgStatus
     property bool imgIsIdenticon: false
 
     // ring settings hints
     readonly property real ringPxSize: Math.max(1.5, root.width / 24.0)
+
+    function isImgSrc(name) {
+        return name.includes("data") || name.toLowerCase().startsWith("http://") || name.toLowerCase().startsWith("https://")
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml:44]:
```qml
return name.startsWith(Constants.dataImagePrefix) || name.toLowerCase().startsWith("http://") || name.toLowerCase().startsWith("https://")
```

File: ui/StatusQ/src/StatusQ/Components/StatusSmartIdenticon.qml [hunk 2]
```diff
@@ -24,8 +24,12 @@ Loader {
         distinctiveColors: Theme.palette.identiconRingColors
     }
 
-    sourceComponent: root.loading ?  loadingComp : (root.asset.isLetterIdenticon || root.asset.name === "") ? letterIdenticon :
-                     !root.asset.isImage ? roundedIcon : roundedImage
+    sourceComponent: {
+        if (root.loading) return loadingComp
+        if (root.asset.isLetterIdenticon || root.asset.name === "") return letterIdenticon
+        if (root.asset.isImage) return roundedImage
+        return roundedIcon
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusSmartIdenticon.qml:31]: yay! :rocket:

File: ui/imports/shared/views/chat/MessageView.qml [hunk 3]
```diff
@@ -602,7 +602,6 @@ Loader {
                         width: 40
                         height: 40
                         name: root.senderIcon || ""
-                        assetSettings.isImage: root.isDiscordMessage || root.senderIcon.startsWith("data")
```
Comment (caybro) [inline ui/imports/shared/views/chat/MessageView.qml:605]: Discord special case dropped?

**Review by caybro — APPROVED** [review]
LGTM, just some questions inline

**Review by jrainville — COMMENTED** [review]
Looks good. It seems way cleaner.

The only thing I want to make sure is do we still update the messages when someone changes their display name or if we put a nickname?

File: ui/imports/shared/views/chat/MessageView.qml (hunk 3 above)
Comment (osmaczko) [reply ui/imports/shared/views/chat/MessageView.qml:605]: Yes, seems not needed. I tested discord messages and contact images are still there.

**Comment by osmaczko** [comment]
> Looks good. It seems way cleaner.
> 
> The only thing I want to make sure is do we still update the messages when someone changes their display name or if we put a nickname?

All cases should be covered the same way as before. I just transform `message.text` on service level rather than in separate modules. I will double check anyway.

File: ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml (hunk 1 above)
Comment (osmaczko) [reply ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml:44]: Good suggestion but can't use it, `Constants` are not part of `StatusQ`. I would need to introduce StatusQ's `Constants` and move it there but I don't think it should be in the scope of this PR.

**Review by jrainville — APPROVED** [review]
Nice

File: ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml (hunk 1 above)
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml:44]: Ah right :/ Still even a hardcoded `startsWith("data:image")` would be better imho

File: ui/imports/shared/views/chat/MessageView.qml (hunk 3 above)
Comment (caybro) [reply ui/imports/shared/views/chat/MessageView.qml:605]: Cool

File: ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml (hunk 1 above)
Comment (osmaczko) [reply ui/StatusQ/src/StatusQ/Core/StatusAssetSettings.qml]: Agreed, improved

**Pull request merged**